File: xsolla-store-sdk/src/main/java/com/xsolla/android/store/entity/request/cart/CartRequestOptions.java
package com.xsolla.android.store.entity.request.cart;

public class CartRequestOptions {

    private String currency = null;
    private String locale = null;

    public String getCurrency() {
        return currency;
    }

    public String getLocale() {
        return locale;
    }

    public Builder create() {
        return new CartRequestOptions().new Builder();
    }

    public class Builder {

        private Builder() {

        }

        public Builder setCurrency(String currency) {
            CartRequestOptions.this.currency = currency;
            return this;
        }

        public Builder setLocale(String locale) {
            CartRequestOptions.this.locale = locale;
            return this;
        }

        public CartRequestOptions build() {
            return CartRequestOptions.this;
        }

    }
}

File: xsolla-store-sdk/src/main/java/com/xsolla/android/store/entity/request/cart/CartRequestOptions.kt
package com.xsolla.android.store.entity.request.cart

class CartRequestOptions {
    var currency = "USD"
    var locale = "en"

    fun create() = Builder()

    inner class Builder {
        fun setCurrency(currency: String): Builder {
            this@CartRequestOptions.currency = currency
            return this
        }

        fun setLocale(locale: String): Builder {
            this@CartRequestOptions.locale = locale
            return this
        }

        fun build() = this@CartRequestOptions
    }
}